recipes/oslo.utils/meta.yaml

```diff
@@ -0,0 +1,61 @@
+{%set name = "oslo.utils" %}
+{%set version = "3.16.0" %}
+{%set compress_type = "tar.gz" %}
+{%set hash_type = "sha256" %}
+{%set hash_val = "109e018da9d95caba79c94935257d2335e5a77e65ebbff218cb9756e746630f1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.{{ compress_type }}
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ compress_type }}
+  {{ hash_type }}: {{ hash_val }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - pbr >=1.8
+
+  run:
+    - python
+    - pbr >=1.6
```

Comment:
The short answer is "because that's what upstream has declared."

setup.py [says](https://github.com/openstack/oslo.utils/blob/3.16.0/setup.py#L28):
```python
setup_requires=['pbr>=1.8'],
```

but requirements.txt [says](https://github.com/openstack/oslo.utils/blob/3.16.0/requirements.txt#L10)
```python
pbr>=1.6 # Apache-2.0
```

The upstream change that declared this 1.6/1.8 requirement is [here](https://review.openstack.org/#/c/224330/), the commit description explains that we need `pbr>=1.8` at setup/build-time to handle comments in marker lines - but there's a weaker version requirement at runtime since we only need to be able to return version and other information from the (more stable) dist-info metadata.

To complicate it further, oslo.utils doesn't actually use pbr at run-time, but the requirement is necessary (aiui) to force setuptools/pbr to install packages in the correct order when a 3rd-party requires both oslo.utils *and* pbr.  See https://bugs.launchpad.net/pbr/+bug/1384919

More generally: Should we be making these sorts of efforts to investigate and double-guess upstream's metadata declarations?  I feel like I've just spent an hour of my life only to reach the unsurprising conclusion that upstream knows more about their software than I do.  @jakirkham, what's the actual issue here?